management-interface.go

```diff
@@ -289,6 +293,301 @@ func NetworkHandler(w http.ResponseWriter, r *http.Request) {
+
+	//get the networkid of the new networks from stdOut
+	scanner := bufio.NewScanner(strings.NewReader(stdOut))
+	scanner.Scan() //skip interface line
+	if scanner.Scan() {
+		line := scanner.Text()
+		networkId, err = strconv.Atoi(line)
+		if err != nil {
+			log.Printf("could not find network id - error %s from stdout %s", err.Error(), stdOut)
+			return -1, err
+		}
+	}
+	return networkId, err
+}
+
+//setWpaNetworkDetails - sets the ssid and password of the specified networkID
+func setWpaNetworkDetails(ssid string, password string, networkId int) error {
+	cmd := exec.Command("wpa_cli")
+	stdin, err := cmd.StdinPipe()
+	if err != nil {
+		log.Printf("error getting stdin pipe from cnd -error %s", err.Error())
+		return err
+	}
+
+	defer stdin.Close()
+	io.WriteString(stdin, fmt.Sprintf("set_network %d ssid \"%s\"\n", networkId, ssid))
+	io.WriteString(stdin, fmt.Sprintf("set_network %d psk \"%s\"\n", networkId, password))
+	io.WriteString(stdin, fmt.Sprintf("enable_network %d\n", networkId))
+	io.WriteString(stdin, "quit\n")
+	var out []byte
+
+	out, err = cmd.CombinedOutput()
+
+	if err != nil {
+		log.Printf("error adding wpa network -error %s", err.Error())
+		return err
+	}
+
+	errOccured := hasErrorOccured(string(out))
+	if errOccured {
+		reloadWpaConfig()
+		err = errors.New("error setting new network")
+		log.Printf(err.Error())
+	}
+	return err
+}
+
+//reloadWpaConfig - executes wpa_cli reconfigure
+func reloadWpaConfig() error {
+	out, err := exec.Command("wpa_cli", "reconfigure").Output()
+	if err != nil {
+		log.Printf("error reloading config - error %s output %s", err.Error(), out)
+		return err
+	}
+
+	errOccured := hasErrorOccured(string(out))
+	if errOccured {
+		err = errors.New("error reloading config")
+		log.Printf(err.Error())
+	}
+	return err
+}
+
+//hasErrorOccured - checks stdout for FAIL text
+func hasErrorOccured(stdOut string) bool {
+	scanner := bufio.NewScanner(strings.NewReader(stdOut))
+	scanner.Scan() //skip the interface line
+	for scanner.Scan() {
+		if scanner.Text() == "FAIL" {
+			log.Printf("error in output %s", stdOut)
+			return true
+		}
+	}
+	return false
+}
+
+//saveWpaConfig - executes wpa_cli save config
+func saveWpaConfig() error {
+	out, err := exec.Command("wpa_cli", "save", "config").Output()
+	if err != nil {
+		log.Printf("error saving config - error %s output %s", err.Error(), out)
+		return err
+	}
+	errOccured := hasErrorOccured(string(out))
+	if errOccured {
+		err = errors.New("error saving config")
+		log.Printf(err.Error())
+	}
+	return err
+}
+
+type wifiNetwork struct {
+	Ssid      string
+	NetworkID int
+}
+
+//parseWpaSupplicantConfig - uses wpa_cli list_networks to get current networks in wpa_supplicant
+func parseWpaSupplicantConfig() ([]wifiNetwork, error) {
+	var out []byte
+	err := error(nil)
+	out, err = exec.Command("wpa_cli", "list_networks").Output()
+	networks := []wifiNetwork{}
+
+	if err != nil {
+		log.Printf("error parsing wpa config" + err.Error())
```

Comment:
This ends up rendering like this:
```
error parsing wpa configexit status 255
```

You probably want something like:
```go
log.Printf("error parsing wpa config: %v", err)
```

The `%v` format specifier will automatically call Error()